### PR TITLE
[BEAM-4809] Build javadoc on :javaPreCommit and :javaPostCommit

### PR DESCRIPTION
**Pull request by alanmyrvold**
Build javadoc on :javaPreCommit and :javaPostCommit to check it builds ok

**Please** add a meaningful description for your change here

-----------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by alanmyrvold** [comment]
Javadoc should be fixed by https://github.com/apache/beam/pull/5972

**Comment by alanmyrvold** [comment]
@boyuanzz

**Comment by lukecwik** [comment]
Do you think it would be better to ensure that the `check` task depends on `javadoc` in each project. This way during the course of a normal build a user would get javadoc failures instead of only getting the failure during the precommit run?

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by boyuanzz** [comment]
Hey Alan, do we have enabled java spotless tasks in precommit and postcommit? If not, do you think we can add them into at least one of the check job?

File: build.gradle [hunk 1]
```diff
@@ -151,6 +151,7 @@ rat {
   excludes = exclusions
 }
 check.dependsOn rat
+check.dependsOn ":beam-sdks-java-javadoc:allJavadoc"
```
Comment (lukecwik) [inline build.gradle]: This means that the root project will check that all javadocs are well formed which is typically something that we don't want to impose when running `check` within the root project.

I think you meant to edit BeamModulePlugin.groovy and add:
`project.check.dependsOn project.javadoc` within `applyJavaNature`

Comment (alanmyrvold) [reply build.gradle]: Done, moved to BeamModulePlugin.groovy

**Comment by alanmyrvold** [comment]
Run Java PreCommit

**Comment by alanmyrvold** [comment]
Run Java PreCommit

**Comment by alanmyrvold** [comment]
Run Seed Job

**Comment by alanmyrvold** [comment]
Run Java PreCommit

**Comment by alanmyrvold** [comment]
Run Java PreCommit

**Comment by udim** [comment]
Run Java PreCommit

**Comment by aaltay** [comment]
What is the state of this PR?

**Comment by alanmyrvold** [comment]
@aaltay It is ready to be merged;
precommits passed including the new javadoc check.

**Comment by pabloem** [comment]
Very nice. Merging.

**Pull request merged**